ansible-runner/context/app/project/roles/mirror_appliance/templates/pg_dump.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+CPU_COUNT=$(nproc 2>/dev/null || echo 1)
+CPU_COUNT=$(( CPU_COUNT > 0 ? CPU_COUNT : 1))
+echo "CPU COUNT: ${CPU_COUNT}"
+
+# clean up postgres tmp/pgdump_dir if previously created
+podman exec quay-postgres rm -rf /tmp/pgdump_dir
+
+# Take pg_dump of postgres container in parallel and store it in tmp/pgdump_dir
+podman exec -it quay-postgres \
+    pg_dump -j "$CPU_COUNT" -Z 0 --no-sync --format=directory \
+    --data-only --column-inserts --no-owner --no-privileges \
```

Comment:
this is dumping the data only, how does the schema structure get exported? If you're able to push/pull after the migration this must not be an issue

Reply:
the schema is already generated as a [pre-step here](https://github.com/quay/mirror-registry/blob/main/ansible-runner/context/app/project/roles/mirror_appliance/tasks/migrate.yaml#L110) before taking the data only dump. 
Yep, push/ pull and mirroring images works both before and after migration.